lib/puppet/util/run_mode.rb

```diff
@@ -38,6 +38,10 @@ def log_dir
         "$vardir/log"
       end
 
+      def conf_dir
+        "$puppetdir/config"
+      end
+
```

Comment:
IIRC there is some extra magic about variable interpolation in settings, where only some settings which are "known" at this stage of the initialization can be interpolated... IIRC @jpartlow got down deep into this at one point, maybe as a result of https://tickets.puppetlabs.com/browse/PUP-1015  ??